### PR TITLE
Add maxWidth support

### PR DESCRIPTION
**Pull request by johnotander**

### CODE REVIEW

File: src/measure.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const bp = breaks(props)
+
+  return n
+    .map(mx(scale))
+    .map(dec('width'))
+    .map(media(bp))
+    .reduce(merge, {})
+}
+
+const mx = scale => n => num(n) ? px(scale[n] || n) : n
```
Comment (johnotander) [inline src/measure.js]: Just realized this should be `em` : )

File: src/measure.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+module.exports = props => {
+  const n = props.measure
+
+  if(!is(n)) return null
+
+  const scale = get(props, 'theme.measures', measures)
+
+  if (!Array.isArray(n)) {
+    return {
+      width: mx(scale)(n)
```
Comment (jxnblk) [inline src/measure.js]: I feel like this should use `max-width` and the utility should be called `maxWidth` – sticking to CSS naming conventions like the rest of (most of??) the library

File: src/constants.js [hunk 3]
```diff
@@ -24,8 +24,15 @@ const fontSizes = [
   72
 ]
 
+const measures = [
+  24,
+  32,
+  48
```
Comment (jxnblk) [inline src/constants.js]: In the next major version, I plan on changing the em-unit breakpoints to be pixel values to keep everything consistent. Would be great to handle this with pixels as well, but still allow em-units to be defined with a string

File: src/measure.js (hunk 2 above)
Comment (johnotander) [reply src/measure.js]: Yeah, `max-width` makes sense to me. I've updated my pr to use the css naming.

File: src/max-width.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const { get, is, px, num, breaks, dec, media, merge } = require('./util')
+const { maxWidths } = require('./constants')
```
Comment (jxnblk) [inline src/max-width.js]: I've been leaning towards fewer (potentially none) default values in styled-system – thinking it's better to work off of raw values, then let users set their own values in their theme – and I think max-widths are one of those things that probably vary widely from project to project

File: README.md [hunk 5]
```diff
@@ -324,6 +325,14 @@ import { boxShadow } from 'styled-system'
 // props.theme.shadows[1]
 ```
 
+### maxWidth (responsive)
+
+```js
+import { measure } from 'styled-system'
```
Comment (jxnblk) [inline README.md]: Typo?

File: src/max-width.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const width = require('./width')
+
+module.exports = props => {
```
Comment (jxnblk) [inline src/max-width.js]: Might be missing something, but is there a reason not to use the `responsiveStyle` utility for this?

Comment (johnpolacek) [reply src/max-width.js]: That's what I do - https://github.com/johnpolacek/styled-system-html/blob/ecfc999347102adabe81dca43e7d5271d29ce66a/src/styles.js#L7 - works great :)

**Pull request merged**